Boards/m5stack-papers3/Source/Configuration.cpp

```diff
@@ -0,0 +1,86 @@
+#include <vector>
+#include <memory>
+#include <Tactility/hal/Configuration.h>
+#include <Tactility/lvgl/LvglSync.h>
+
+#include "devices/Display.hpp"
+#include "devices/Power.hpp"
+#include "devices/SdCard.hpp"
+
+bool initBoot();
+
+using namespace tt::hal;
+
+static std::vector<std::shared_ptr<Device>> createDevices() {
+    return {
+        createPower(),
+        createDisplay(),
+        createSdCard()
+    };
+}
+
+extern const Configuration hardwareConfiguration = {
+    .initBoot = initBoot,
+    .createDevices = createDevices,
+    .i2c = {
+        i2c::Configuration {
+            .name = "Internal",
+            .port = I2C_NUM_0,
+            .initMode = i2c::InitMode::ByTactility,
+            .isMutable = false,
+            .config = (i2c_config_t) {
+                .mode = I2C_MODE_MASTER,
+                .sda_io_num = GPIO_NUM_41,
+                .scl_io_num = GPIO_NUM_42,
+                .sda_pullup_en = true,
+                .scl_pullup_en = true,
+                .master = {
+                    .clk_speed = 400000
+                },
+                .clk_flags = 0
+            }
+        },
+        i2c::Configuration {
+            .name = "External",
+            .port = I2C_NUM_1,
```

Comment:
Assuming this is the grove port, can you add a UART config for these pins too? You can compare with one of the other M5Stack device configs.
That would allow for peripherals like GPS to be connected to the Grove port. (the M5Stack GPS is actually supported!)

This is the reason that InitMode::Disabled is set: because the port is usable both as UART and as I2C.